### PR TITLE
Log SCTs via RequestLog API when they are about to be issued

### PR DESCRIPTION
**Pull request by Martin2112**

### CODE REVIEW

File: trillian/ctfe/requestlog.go [hunk 1]
```diff
@@ -64,6 +64,12 @@ type RequestLog interface {
 	// LeafHash will be called once for get proof by hash requests with the
 	// requested hash value (if the parameters parse correctly).
 	LeafHash(context.Context, []byte)
+	// SCT will be called once when the server is about to issue an SCT to a
+	// client. This need not be called if the submission process fails before an
+	// SCT could be presented to a client, even if this is unrelated to
+	// the validity of the submitted chain. The SCT bytes will be in TLS
+	// serialized format.
+	SCT(context.Context, []byte)
```
Comment (daviddrysdale) [inline trillian/ctfe/requestlog.go]: Maybe `IssueSCT`?

File: trillian/ctfe/requestlog.go [hunk 2]
```diff
@@ -64,6 +64,12 @@ type RequestLog interface {
 	// LeafHash will be called once for get proof by hash requests with the
 	// requested hash value (if the parameters parse correctly).
 	LeafHash(context.Context, []byte)
+	// SCT will be called once when the server is about to issue an SCT to a
+	// client. This need not be called if the submission process fails before an
```
Comment (AlCutter) [inline trillian/ctfe/requestlog.go]: /need not/should not/?
i.e. "only set this when the associated leaf is known to have been successfully queued and the intention is to issue the SCT"?

File: trillian/ctfe/requestlog.go [hunk 3]
```diff
@@ -125,6 +131,11 @@ func (dlr *DefaultRequestLog) LeafHash(_ context.Context, lh []byte) {
 	glog.V(vLevel).Infof("RL: LeafHash: %s", hex.EncodeToString(lh))
 }
 
+// SCT logs an SCT that will be issued to a client.
+func (dlr *DefaultRequestLog) SCT(_ context.Context, sct []byte) {
+	glog.V(vLevel).Infof("RL: Issuing SCT: %s", hex.EncodeToString(sct))
```
Comment (daviddrysdale) [inline trillian/ctfe/requestlog.go]: Wouldn't a format specifier of %x do the trick without `hex.EncodeToString` ?

File: trillian/ctfe/requestlog.go (hunk 1 above)
Comment (Martin2112) [reply trillian/ctfe/requestlog.go]: Done.

File: trillian/ctfe/requestlog.go (hunk 3 above)
Comment (Martin2112) [reply trillian/ctfe/requestlog.go]: Done.

**Pull request merged**